### PR TITLE
CI, Harmonization of tests, Testing of Chef 13 and Chef 12

### LINKED ISSUE #174: Chef 13 support
It would be cool if we would validate chef 12 and chef 13 in the CI

### PR DESCRIPTION
**Pull request by artem-sidorenko**
- Harmonization of test and CI setup with os-hardening
 - Run tests per default on latest chef version and do some tests with Chef 12

@chris-rock @atomic111 I took a similar approach like in os-hardening, `.kitchen.yml` contains now vagrant setup and some parameters, `.kitchen.dokken.yml` contains configs for dokken/docker.

I had some problems with dokken: because of systemd thing, privileged container and some error the systemd of my host was influenced by the container process and broke. Since then I'm only using the vagrant way. I do not know your setups, but I hope this change is Okay. If not, let us discuss the alternatives

Fixes #174, #177

### CODE REVIEW

**Comment by artem-sidorenko** [comment]
Red tests of opensuse leap should be fixed by https://github.com/dev-sec/ssh-baseline/pull/88

File: .kitchen.dokken.yml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    pid_one_command: /usr/lib/systemd/systemd
+
+suites:
+- name: default
+  run_list:
+  - recipe[test]
+  - recipe[ssh-hardening]
+  verifier:
+    inspec_tests:
+      - https://github.com/dev-sec/tests-ssh-hardening
```
Comment (atomic111) [inline .kitchen.dokken.yml]: can you change the link to https://github.com/dev-sec/ssh-baseline

File: .kitchen.yml [hunk 2]
```diff
@@ -78,3 +32,27 @@ suites:
   verifier:
     inspec_tests:
       - https://github.com/dev-sec/tests-ssh-hardening
```
Comment (atomic111) [inline .kitchen.yml]: can you change the link to https://github.com/dev-sec/ssh-baseline

File: .kitchen.yml [hunk 3]
```diff
@@ -78,3 +32,27 @@ suites:
...
+  - recipe[test]
+  - recipe[ssh-hardening]
+  attributes:
+    ssh-hardening:
+      ssh:
+        server:
+          use_pam: false
+  verifier:
+    inspec_tests:
+      - https://github.com/dev-sec/tests-ssh-hardening
```
Comment (atomic111) [inline .kitchen.yml]: can you change the link to https://github.com/dev-sec/ssh-baseline

File: .kitchen.dokken.yml (hunk 1 above)
Comment (artem-sidorenko) [inline .kitchen.dokken.yml]: good catch, thanks! :)

**Pull request merged**